rust/index/src/fulltext/types.rs

```diff
@@ -92,60 +93,71 @@ impl<'me> FullTextIndexWriter<'me> {
 
     async fn populate_frequencies_and_posting_lists_from_previous_version(
         &self,
-        token: &str,
+        tokens: &Vec<Token>,
     ) -> Result<(), FullTextIndexError> {
         let mut uncommitted_frequencies = self.uncommitted_frequencies.lock().await;
-        match uncommitted_frequencies.get(token) {
-            Some(_) => return Ok(()),
-            None => {
-                let frequency = match &self.full_text_index_reader {
-                    // Readers are uninitialized until the first compaction finishes
-                    // so there is a case when this is none hence not an error.
-                    None => 0,
-                    Some(reader) => match reader.get_frequencies_for_token(token).await {
-                        Ok(frequency) => frequency,
-                        // New token so start with frequency of 0.
-                        Err(_) => 0,
-                    },
-                };
-                uncommitted_frequencies
-                    .insert(token.to_string(), (frequency as i32, frequency as i32));
+        for token in tokens {
+            match uncommitted_frequencies.get(&token.text) {
+                Some(_) => return Ok(()),
```

Comment:
Isn't this wrong? We should continue here to the next token instead of returning